auth/client/src/main/java/org/wildfly/security/credential/source/OAuth2CredentialSource.java

```diff
@@ -61,7 +61,6 @@
  *
  * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
  */
-@Deprecated
```

Comment:
I think we still need to leave this as deprecated until we come up with an alternative solution.  We probably need a Blocker issue to flag this is still not complete.

Reply:
I can redeprecate it, but I'm not sure if it would be possible to move it without modifying the class due to the circular dependency for auth.client it would cause. If I do redeprecate it, should I add a comment explaining why it hasn't been moved is due to the circular dependency?

Reply:
Yes please keep it deprecated - we may need a different solution to create the alternative but some solution will be needed.

Reply:
Added the deprecation back (since that was all the ELY-1724 commit did, I removed that commit)